core/state/access_witness.go

```diff
@@ -237,7 +237,7 @@ func (aw *AccessWitness) TouchCodeChunksRangeAndChargeGas(contractAddr []byte, s
 	// reason that we do not need the last leaf is the account's code size
 	// is already in the AccessWitness so a stateless verifier can see that
 	// the code from the last leaf is not needed.
-	if (codeLen == 0 && size == 0) || startPC > codeLen {
+	if size == 0 || startPC > codeLen {
```

Comment:
Took me a while to remember what was going on here, but the fact is that there is an issue if the code length is 0. If the code length is 0, though, this should be captured by `startPC >= codeLen`

```suggestion
	if size == 0 || startPC >= codeLen {
```